pkgs/shells/fish/default.nix

```diff
@@ -109,12 +109,8 @@ let
     '';
 
     patches = [
-      # Fixes "Integer 243 in '243 (243)' followed by non-digit" error with systemctl completion.
-      # https://github.com/fish-shell/fish-shell/issues/5689 (will be included in fish 3.1.0)
-      (fetchpatch {
-        url = "https://github.com/fish-shell/fish-shell/commit/c6ec4235136e82c709e8d7b455f7c463f9714b48.patch";
-        sha256 = "02m6pkhhx6y21csydznsxkbpnwhcpzyz99xgd9ryh7s03v7wbigw";
-      })
+      # Fixes compilation on old Apple SDKs
+      ./old-sdk-compilation.patch
```

Comment:
Can we please use ```fetchpatch``` from https://github.com/fish-shell/fish-shell/pull/6602 instead of adding the patch in nixpkgs?

Reply:
The only reason I went with adding the patch is because of above:

> Well, I suppose fetching the PR as a patch opens us up to hash changes. I'm not sure what the best solution is prior to the PR being merged; we could fetch the commit, but then if the PR's history is edited the commit isn't guaranteed to stay.

If you still think `fetchpatch` should be used, I'll do so.

Reply:
I decided to `fetchpatch` the commit itself (should hopefully be more static).

Reply:
The PR has been merged (and is slated for 3.1.1) so fetching the commit is definitely safe (fetching the PR is safe at this point too, but fetching the commit is just as good).